pandas/io/pickle.py

```diff
@@ -15,12 +16,16 @@ def to_pickle(obj, path):
     obj : any object
     path : string
         File path
+    compression : {'infer', 'gzip', 'bz2', 'xz', None}, default 'infer'
+        .. versionadded:: 0.19.2
     """
-    with open(path, 'wb') as f:
+    inferred_compression = _infer_compression(path, compression)
+    f, fh = _get_handle(path, 'wb', compression=inferred_compression, is_text=False)
+    with f:
         pkl.dump(obj, f, protocol=pkl.HIGHEST_PROTOCOL)
```

Comment:
I think we should close the file handles in `fh` explicitly, instead of closing `f` by using it in a `with` statement